troubleshooting.md

```diff
@@ -0,0 +1,551 @@
+# Troubleshooting Guide for BM IPI Deployments
+
+* [Introduction](#introduction) 
+
+* [Bootstrap VM related issues](#bootstrap-vm-related-issues)
+
+* [Node Related Issues](#node-related-issues)
+
+* [Registry Issues](#registry-issues) 
+
+* [IPv6 Disconnected Install Troubleshooting](#ipv6-disconnected-install-troubleshooting)
+
+* [Misc Issues](#misc-issues) 
+
+# Introduction
+ 
+**BM installation for OCP4:** While deploying BM version of Openshift there are some mandatory requirements which needs to be followed, otherwise it may fail. This guide attempts to cover some of the common scenarios of such events fixing which might help you get a clean cluster. 
+
+**Pre-requisites:** 
+
+First and foremost the BM installations follows a general architecture:
+* provisioner node/worker node(which can be converted after installation). 
+* master node 
+* worker node
+
+**Provisoner node:**
+
+This node acts as a helper node which creates a Bootstrap VM which boots up a temporary control plane and helps the master boot up using an ironic conductor. After all the components are generated the VM transfers the original control plane to master node and shuts itself down. 
+
+* Make sure you can reach `172.22.0.0/24` network from your provisioner node.
+
+```
+[kni@provisioner ~]$ ping -c3 172.22.0.1
+PING 172.22.0.1 (172.22.0.1) 56(84) bytes of data.
+64 bytes from 172.22.0.1: icmp_seq=1 ttl=64 time=0.049 ms
+64 bytes from 172.22.0.1: icmp_seq=2 ttl=64 time=0.041 ms
+64 bytes from 172.22.0.1: icmp_seq=3 ttl=64 time=0.030 ms
+```
+This will ensure you have proper access to the DHCP network which will then provide internal IP's to all the cluster nodes. 
+
+* Make sure you can reach the VIP's of API, NS and APPS before starting the deployments. 
+
+* Create an install-config file as [Install-config](https://gist.github.com/hardys/30809af4d2d6f89354cec60241b4883d)
```

Comment:
The odd parts with this is they don't troubleshoot anything. It's more of your telling the user that this should be set. The troubleshooting guide should focus on a problem and how to solve that problem. The above is just general guidelines which is already covered in the install-steps

Reply:
This was just a pre-req that needed to avoid further errors during deployment, but I will remove it as suggested.